static/geolocation.js

```diff
@@ -1,48 +1,101 @@
-(function() {
-  var startTime = new Date().getTime();
+window.position = {
+  pos: null,
+  error: null,
+  timing: null
+}
 
-  var quietTimeoutSeconds = 20;
+window.retrieveGeolocation = function retrieveGeolocation(pos) {
+  window.position.pos = pos;
+}
 
-  window.position = {
-    pos: null,
-    error: null,
-    timing: null
-  }
+window.retrieveGeolocationError = function retrieveGeolocationError(error) {
+  window.position.error = error;
+}
 
-  window.retrieveGeolocation = function retrieveGeolocation(pos) {
-    window.position.pos = pos;
-  }
+window.retrieveGeolocationTiming = function retrieveGeolocationTiming(timing) {
+  window.position.timing = timing;
+}
 
-  window.retrieveGeolocationError = function retrieveGeolocationError(error) {
-    window.position.error = error;
+// for prompt/positioning detection
+function clearTimeout2(){
+  if (timeout2 !== null) {
+    clearTimeout(timeout2);
+    timeout2 = null;
   }
+}
 
-  window.retrieveGeolocationTiming = function retrieveGeolocationTiming(timing) {
-    window.position.timing = timing;
-  }
+let timeout2;
+
+function startPositioning() {
+  var startTime = new Date().getTime();
+  var quietTimeoutSeconds = 20;
+
+  //timeout for prompt/positioning detection
+  if(navigator.permissions !== undefined) {
+    timeout2 = setInterval(function () {
```

Comment:
https://developer.mozilla.org/en-US/docs/Web/API/PermissionStatus/onchange could be used instead